<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        try{
            setTimeout(() => { throw new Error('Error!'); }, 1000);
        } catch(e){
            console.error('에러를 캐치하지 못한다.', e);
        }
        
        const func = async () => {
            try{
                await new Promise((resolve, reject) => {
                    setTimeout(() => reject(new Error('Error!')), 1000);
                });
            } catch(e){
                console.error('에러를 캐치한다.', e);
            }
        }
        console.log('func() 호출 전');
        try{
            func();
        } catch(e){
            console.error('에러를 캐치한다.');
        }
        console.log('func() 호출 후');
    </script>
</body>
</html>